src/model/UserphraseModel.cpp

```diff
@@ -238,11 +238,11 @@ const Userphrase *UserphraseModel::getUserphrase(const QModelIndex& idx)
 
 void UserphraseModel::undo()
 {
-    if (!userphrase_tmp_.empty()) {
```

Comment:
@jserv : If I rewrite the insert method with a flag to control the operation to "shadow", it will affect the run-test file which also use the insert function. Is it also fine?